system/jhtree/jhtree.cpp

```diff
@@ -2742,20 +2837,14 @@ const CJHTreeNode *CNodeCache::getCachedNode(const INodeLoader *keyIndex, unsign
     try
     {
         //Move the atomic increments out of the critical section - they can be relatively expensive
-        if (likely(alreadyExists))
+        if (likely(ctx))
         {
-            if (ctx) ctx->noteStatistic(hitStatId[cacheType], 1);
-        }
-        else
-        {
-            if (ctx) ctx->noteStatistic(addStatId[cacheType], 1);
+            if (unlikely(alreadyExists))
+                ctx->noteStatistic(hitStatId[cacheType], 1);
```

Comment:
unlikely() is correct - because the code only reaches this point if there was a match in the cache, but the node associated with that entry has not been loaded yet.
I will add a comment to clarify why.

Reply:
all good.